src/Data/Tree/Binary/Relation/Unary/All/Properties.agda

```diff
@@ -12,15 +12,24 @@ open import Level
 open import Data.Tree.Binary as Tree using (Tree; leaf; node)
 open import Data.Tree.Binary.Relation.Unary.All
 open import Relation.Unary
+open import Function
 
 private
   variable
-    a b p q : Level
+    a b c d p q : Level
     A : Set a
     B : Set b
+    C : Set c
+    D : Set d
+    P : C → Set p
+    Q : D → Set q
 
-module _ {P : B → Set p} where
+map⁺ : (f : A → C) → (g : B → D) → ∀[ All (f ⊢ P) (g ⊢ Q) ⇒ Tree.map f g ⊢ All P Q ]
```

Comment:
Hmm, this is another one of those cases where I think we might be using too much notation. I'm really struggling to interpret the exact type of this. Perhaps expanding at least some of the notation out might make it more usable?

Reply:
I agree though I don't think it is more confusing than before

Reply:
I've changed this though I don't know if it is any better

Reply:
Looks ever so slightly better to me...